quixstreams/app.py

```diff
@@ -710,6 +721,32 @@ def clear_state(self):
         """
         self._state_manager.clear_stores()
 
+    def source(self, source: BaseSource, topic: Optional[Topic] = None) -> Topic:
+        """
+        Add a source to the application.
+
+        See :class:`quixstreams.sources.base.BaseSource` for more details.
+
+        :param source: a :class:`quixstreams.sources.BaseSource` instance
+        :param topic: the :class:`quixstreams.models.Topic` instance the source will produce to
+            Default: the source default
+        """
+        if not topic:
+            topic = self._topic_manager.source_topic(source)
+
+        producer = RowProducer(
+            broker_address=self._broker_address,
+            extra_config=self._producer_extra_config,
+            flush_timeout=self._consumer_extra_config.get(
+                "max.poll.interval.ms", _default_max_poll_interval_ms
+            )
+            / 1000,  # convert to seconds
```

Comment:
Readability suffers here a little bit. I suggest breaking it down into explicit stages:

```python
max_poll_interval_ms = self._consumer_extra_config.get(
    "max.poll.interval.ms", _default_max_poll_interval_ms
)
flush_timeout_seconds = max_poll_interval_ms / 1000
producer = RowProducer(
    ...,
    flush_timeout=flush_timeout_seconds,
)